File: presentation/src/main/java/example/com/presentation/di/modules/ActivityModule.java
package example.com.presentation.di.modules;

import android.app.Activity;
import android.content.Context;

import dagger.Module;
import dagger.Provides;
import example.com.presentation.di.annotations.ActivityContext;

/**
 * Created by Mohammad Jafarzadeh Rezvan on 09/11/2018.
 */

@Module
public class ActivityModule {

    private Activity mActivity;

    public ActivityModule(Activity activity) {
        mActivity = activity;
    }

    @Provides
    Activity provideActivity() {
        return mActivity;
    }

    @Provides
    @ActivityContext
    Context providesContext() {
        return mActivity;
    }
}

File: presentation/src/main/java/example/com/presentation/di/modules/ActivityModule.kt
package example.com.presentation.di.modules

import android.app.Activity
import android.content.Context

import dagger.Module
import dagger.Provides
import example.com.presentation.di.ActivityContext

/**
 * Created by Mohammad Jafarzadeh Rezvan on 09/11/2018.
 */

@Module
class ActivityModule(private val mActivity: Activity) {

    @Provides
    fun provideActivity(): Activity = mActivity

    @Provides
    @ActivityContext
    fun providesContext(): Context = mActivity
}
